### PR TITLE
api: swagger: fix documentation for image push endpoint

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/issues/42053

The documentation was incorrect and didn't properly document the use of tags;

- Image push currently only accepts an image-name, not an ID / digest.
- When giving a name, it's expected to be without tag included; when including a tag, it is ignored.
- The tag parameter is required when pushing a single image (i.e., it does not default to "latest"); omitting the tag parameter will push all tags of the given image.

**- Description for the changelog**

```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Opening as draft, because I'm not really sold on the wording 😂 - once we get to better wording, I can update the versioned copies that are used in the docs.

cc @dvdksn perhaps you have good suggestions

File: api/swagger.yaml [hunk 1]
```diff
@@ -9231,12 +9231,24 @@ paths:
       parameters:
         - name: "name"
           in: "path"
-          description: "Image name or ID."
+          description: |
+            Name of the image to push, for example, `registry.example.com/myimage`.
```
Comment (dvdksn) [inline api/swagger.yaml:9235]:
```suggestion
            Name of the image to push. For example, `registry.example.com/myimage`.
```

File: api/swagger.yaml [hunk 2]
```diff
@@ -9231,12 +9231,24 @@ paths:
...
+            The image must be present in the local image store and tagged with
+            the given name.
```
Comment (dvdksn) [inline api/swagger.yaml:9237]: maybe avoid "tagged" with here? 😅 

```suggestion
            The image must be present in the local image store with the same name.
```

File: api/swagger.yaml [hunk 3]
```diff
@@ -9231,12 +9231,24 @@ paths:
...
+
+            The name should be provided without tag; if a tag is provided, it
+            is ignored (for example, `registry.example.com/myimage:latest` is
+            considered equivalent to ``registry.example.com/myimage`).
```
Comment (dvdksn) [inline api/swagger.yaml:9241]:
```suggestion
            is ignored. For example, `registry.example.com/myimage:latest` is
            equivalent to `registry.example.com/myimage`.
```

**Comment by thaJeztah** [comment]
Updated; also updated the versioned files used in the docs

**Review by dvdksn — APPROVED** [review]
LGTM

**Review by vvoland — APPROVED** [review]

**Pull request merged**